### PR TITLE
fix(NationalIdentityNumber): use `onBlurValidator` instead of `validator`

### PR DESCRIPTION
**Pull request by langz**
Just to align with validation of organization number https://github.com/dnbexperience/eufemia/pull/3976

### CODE REVIEW

File: packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx [hunk 1]
```diff
@@ -183,46 +179,42 @@ describe('Field.NationalIdentityNumber', () => {
...
         validateInitially
         validate={false}
       />
     )
 
-    await expectNever(() => {
-      // Can't just waitFor and expect not to be in the document, it would approve the first render before the error might appear async.
-      expect(screen.queryByRole('alert')).toBeInTheDocument()
-    })
+    fireEvent.blur(document.querySelector('input'))
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx:205]: Here we use `validator` with fire blur event. Is that on purpose?

File: packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx [hunk 2]
```diff
@@ -231,16 +223,16 @@ describe('Field.NationalIdentityNumber', () => {
...
         validate={false}
         validator={customValidator}
       />
     )
-    await expectNever(() => {
-      // Can't just waitFor and expect not to be in the document, it would approve the first render before the error might appear async.
-      expect(screen.queryByRole('alert')).toBeInTheDocument()
-    })
+
+    fireEvent.blur(document.querySelector('input'))
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx:233]: Here we use `validator` with fire blur event. Is that on purpose?

File: packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx [hunk 3]
```diff
@@ -378,20 +383,24 @@ describe('Field.NationalIdentityNumber', () => {
...
       render(
         <Field.NationalIdentityNumber
           validator={customValidator}
           validateInitially
+          validateUnchanged
           value={id}
         />
       )
+
+      fireEvent.blur(document.querySelector('input'))
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx:402]: Here we use `validator` with fire blur event. Is that on purpose?

File: packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx [hunk 4]
```diff
@@ -423,9 +436,13 @@ describe('Field.NationalIdentityNumber', () => {
           <Field.NationalIdentityNumber
             validator={customValidator}
             validateInitially
+            validateUnchanged
             value={fnr}
           />
         )
+
+        fireEvent.blur(document.querySelector('input'))
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx:444]: Here we use `validator` with fire blur event. Is that on purpose?

**Review by tujoworker — APPROVED** [review]
Looks good. Just unsure about the validator vs event simulation.

File: packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx (hunk 2 above)
Comment (langz) [reply packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx:233]: No, it follows the same "pattern" as done in organization number.
I can fix this here now :)

Comment (langz) [reply packages/dnb-eufemia/src/extensions/forms/Field/NationalIdentityNumber/__tests__/NationalIdentityNumber.test.tsx]: Updated in the latest commit :)
https://github.com/dnbexperience/eufemia/pull/3982/commits/f9035fcd592c1aafdee959690d7b8e998fe7d357

**Comment by langz** [comment]
> Looks good. Just unsure about the validator vs event simulation.

Updated in this commit https://github.com/dnbexperience/eufemia/commit/f9035fcd592c1aafdee959690d7b8e998fe7d357

**Pull request merged**